### ISSUE TITLE
Test failure with shared classes on NFS filesystem

### ISSUE THREAD

**Issue by JamesKingdon** [eclipse/openj9#1340]
One of the tests in testSCCMLTests2 fails with a message "Persistent cache cannot be located on a networked file system". Possibly the problem is the `cacheDir=. ` option.

```
        Testing: Test 116: utilities test: iterate and destroy cache when running with different cacheDir option
        Test start time: 2018/03/04 07:31:56 Eastern Standard Time
        Running command: "/cardhu5/seagate/sdks/jdk/bin/java"  -Xjit -Xgcpolicy:gencon -Xnocompressedrefs -Xrs  -cp /cardhu5/seagate/arm/tests/openj9/test/TestConfig/scripts/testKitGen/../../../../jvmtest/VM_Test/VM_Test.jar -Xshareclasses:name=ShareClassesCMLTests,cacheDir=. com.ibm.j9.sharedCacheAPI.tests.SharedUtilsTest
        Time spent starting: 5 milliseconds
        Time spent executing: 1140 milliseconds
        Test result: FAILED
         [OUT] read Processor   : AArch64 Processor rev 4 (aarch64)
         [OUT]
         [ERR] JVMSHRC287E Persistent cache cannot be located on a networked file system. Either select "nonpersistent" or a different cacheDir.
         [ERR] JVMJ9VM015W Initialization error for library j9shr29(11): JVMJ9VM009E J9VMDllMain failed
         [ERR] Error: Could not create the Java Virtual Machine.
         [ERR] Error: A fatal exception has occurred. Program will exit.

```

**Comment by JamesKingdon** [comment]
May be arm specific. Windows uses '.' for some of these tests, so possibly the code that detects linux vs windows isn't working on arm.

**Comment by hangshao0** [comment]
Probably the current directory is on a network drive. CacheDir cannot be on a network drive.

**Comment by JamesKingdon** [comment]
Yes, the current directory is on NFS. We must be able to run tests on a shared drive, surely? I see code that looks like it picks a different directory for 'linux' vs windows - is that part of the picture?

**Comment by hangshao0** [comment]
Looks like we cannot run this test on a NFS for the time being. We might either pass in another variable pointing to a local dir to "cacheDir=" for this test or change the current test root to a local dir. @llxia , @renfeiw , any comments ?

**Comment by llxia** [comment]
Could you run this test by passing REPORTDIR=/local/dir in make?

**Comment by llxia** [comment]
Below command works for me.
```
make _testSCCMLTests2_1 REPORTDIR=/local/dir
```

**Comment by JamesKingdon** [comment]
OK, that's good to know - thanks.

I think we will need to automate this though. Smaller ARM based development boards typically have their local file system on a micro sd card which are slow and have a limited number of write cycles. We don't want to push any more of the file activity from the test bucket onto there than we have to. Is REPORTDIR specific to the shared class cache tests or more general? Ideally we need a variable that can be set just to specify the directory to use for the scc.

**Comment by llxia** [comment]
REPORTDIR is not specific to shared class cache tests. It can be used for all tests.
I think we can introduce a variable to allow users to set their own cacheDir.
